### ISSUE TITLE
Test_Executable() fails with unified bin hiearchy ("line 41: Expected '/usr/sbin/cat' but got '/usr/bin/cat'")

### ISSUE THREAD

**Issue by jonathonf** [vim/vim#7579]
**Describe the bug**

`Test_Executable()` from `test_functions.vim` fails when `/usr/sbin` is a symlink to `/usr/bin`. This unification of the bin hierarchy is present on Arch Linux (but likely present on other newer or development stream distros too).

**To Reproduce**

Compile vim using https://github.com/archlinux/svntogit-packages/tree/packages/vim/trunk and bumping the `pkgver` to 8.2.2249, e.g.:

```
git clone --depth=1 -b packages/vim https://github.com/archlinux/svntogit-packages.git vim
cd vim/trunk
sed -i 's|^pkgver=.*|pkgver=8.2.2249|;s|^pkgrel=.*|pkgrel=0|' PKGBUILD
makepkg
```

**Expected behavior**

Test should see `/usr/sbin/cat` and `/usr/bin/cat` as equivalent and pass.

**Screenshots**

```
Failures:
        From test_functions.vim:
        Found errors in Test_Executable():
        command line..script /build/src/vim-8.2.2249/src/testdir/runtest.vim[468]..function RunTheTest[39]..Test_Executable line 41: Expected '/usr/sbin/cat' but got '/usr/bin/cat'
```

**Environment (please complete the following information):**
 - Vim version: 8.2.2249
 - OS: Arch Linux x86_64
 - Terminal: n/a (builds in chroot)

**Comment by brammool** [comment]
Weird.  So what is the output of:
cd
echo exepath('cat')
cd /
echo exepath('cat')
echo exepath('bin/cat')
echo exepath('sbin/cat')

**Comment by jonathonf** [comment]
Running inside a build chroot:

```
:cd
/root
:echo exepath('cat')
/usr/sbin/cat
:cd /
/
:echo exepath('cat')
/usr/sbin/cat
:echo exepath('bin/cat')
/usr/bin/cat
:echo exepath('sbin/cat')
/usr/bin/cat
```

Path within the chroot:

```
$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by brammool** [comment]
> This is inside a build chroot:
> 
>     :cd
>     /root
>     :echo exepath('cat')
>     /usr/sbin/cat
>     :cd /
>     /
>     :echo exepath('cat')
>     /usr/sbin/cat
>     :echo exepath('bin/cat')
>     /usr/bin/cat
>     :echo exepath('sbin/cat')
>     /usr/bin/cat

So the weird thing is that exepath('cat') returns "/usr/sbin/cat",
while exepath('sbin/cat') returns "/usr/bin/cat".
The test actually uses exepatch('usr/sbin/cat'), I assume that also
results in "/usr/bin/cat".

Please try this change to src/testdir/test_functions.vim:

--- ../../vim-8.2.2253/src/testdir/test_functions.vim  2020-12-28 12:56:54.179617297 +0100
+++ testdir/test_functions.vim  2020-12-31 14:33:15.009360805 +0100
@@ -1288,7 +1288,13 @@
     " check that the relative path works in /
     lcd /
     call assert_equal(1, executable(catcmd))
-    call assert_equal('/' .. catcmd, catcmd->exepath())
+    let result = catcmd->exepath()
+    " when using chroot looking for sbin/cat can return bin/cat, that is OK
+    if catcmd =~ '\<sbin\>' && result =~ '\<bin\>'
+      call assert_equal('/' .. substitute(catcmd, '\<sbin\>', 'bin', ''), result)
+    else
+      call assert_equal('/' .. catcmd, result)
+    endif
     bwipe
   else
     throw 'Skipped: does not work on this platform'


-- 
Over the years, I've developed my sense of deja vu so acutely that now
I can remember things that *have* happened before ...

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by jonathonf** [comment]
> The test actually uses exepatch('usr/sbin/cat'), I assume that also results in "/usr/bin/cat".

Yes indeed:

```
:cd /
:echo exepath('usr/bin/cat')
/usr/bin/cat
:echo exepath('usr/sbin/cat')
/usr/bin/cat
```
> Please try this change to src/testdir/test_functions.vim:

This patch allows the test to complete successfully! :partying_face:

**Comment by lacygoill** [comment]
Fixed by [8.2.2259](https://github.com/vim/vim/releases/tag/v8.2.2259).
